mscore/scoreview.cpp

```diff
@@ -1206,6 +1206,13 @@ void ScoreView::paint(const QRect& r, QPainter& p)
             if (!ss)
                   return;
 
+            if (!ss->enabled())
+                  ss = ss->next1MMenabled();
+            if (es && !es->enabled())
+                  es = es->prev1MMenabled();
+            if (es && ss->tick() > es->tick())  // start after end?
+                  return;
```

Comment:
Returning from here doesn't look good, at least it misses the `restore()` call to `QPainter`:
https://github.com/musescore/MuseScore/blob/6eb70796d6ed91d306083dbf43acd100feb60379/mscore/scoreview.cpp#L1332

But I see the similar return statements below in this function. Could it make sense to extract range selection drawing routine to a separate function so it would be easier to handle such cases correctly? Though it can probably be done in a separate patch.

Reply:
Agreed.  FWIW, there is probably no case where that statement would actually execute, I did it to be safe.  The selection would have to be pretty messed up this to trigger, though.

Reply:
(agreed it can probably be done in separate patch ::-) )

Reply:
OK, then let's leave it for later indeed.